api/edge_api/identities/traits/views.py

```diff
@@ -0,0 +1,78 @@
+import marshmallow
+from drf_yasg2.utils import swagger_auto_schema
+from flag_engine.api.schemas import APITraitSchema
+from flag_engine.identities.builders import build_identity_dict
+from rest_framework import status, viewsets
+from rest_framework.exceptions import NotFound, ValidationError
+from rest_framework.response import Response
+
+from edge_api.identities.views_mixins import GetIdentityMixin
+from environments.identities.models import Identity
+
+from .permissions import EnvironmentIdentityTraitsPermission
+from .serializers import EdgeIdentityTraitsResponseSerializer
+
+trait_schema = APITraitSchema()
+
+
+class EdgeIdentityTraitsViewSet(viewsets.ModelViewSet, GetIdentityMixin):
+    lookup_field = "trait_key"
```

Comment:
Oh, yeah, that's a good point  - I guess the only option is to pass it as post data?

Reply:
I think that we should simplify all of this to 2 endpoints: 

```
GET /edge-identities/:uuid/edge-traits

 + Response 

    [
        {"trait_key": "foo", "trait_value": "bar"}
    ]

PUT /edge-identities/:uuid:edge-traits

 + Request

    [
        {"trait_key": "foo", "trait_value": "bar"},
        {"trait_key": "new", "trait_value": "new"}
    ]

 + Response

    [
        {"trait_key": "foo", "trait_value": "bar"},
        {"trait_key": "new", "trait_value": "new"}
    ]
```

The PUT endpoint would just blindly set the traits that the FE sends. So we'd end up with something like: 

```python
identity = get_identity()
schema = TraitSchema()

try:
    traits = schema.load(data=request.data, many=True)
except marshmallow.ValidationError:
    ...

identity.identity_traits = traits
identity.save()
return schema.dump(identity.identity_traits, many=True)
```

We could probably just do this with actions on the existing edge identity viewset (which probably shouldn't be a ModelViewSet either but that's a separate issue) rather than a whole new viewset. What do you think?

Reply:
> which probably shouldn't be a ModelViewSet either but that's a separate issue


Nah, I don't think that's true, we use quite a few actions defined in `modelViewSet` like `.retrieve(), .create(), .update()`

Reply:
We do, yes, but we overwrite pretty much all of their functionality. I guess it's a restriction of working with django that it kind of expects everything to relate to a database model. It's fine for now but I do wonder if the code would be slightly neater if we just used `APIView`s for example.